sdk/tests/test_rapid.py

```diff
@@ -408,3 +409,95 @@ def update_subject_permissions_failure(self, requests_mock: Mocker, rapid: Rapid
+        assert res == mocked_response
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_create_user_failure(self, requests_mock: Mocker, rapid: Rapid):
+        mocked_response = {"data": "dummy"}
+        requests_mock.post(f"{RAPID_URL}/user", json=mocked_response, status_code=400)
+        with pytest.raises(SubjectAlreadyExistsException):
+            rapid.create_user("user", "user@user.com", ["READ_ALL"])
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_delete_user_success(self, requests_mock: Mocker, rapid: Rapid):
+        mocked_response = {
+            "username": "user",
+            "user_id": "xxx-yyy-zzz"
+        }
+        requests_mock.delete(
+            f"{RAPID_URL}/user", json=mocked_response, status_code=200
+        )
+        res = rapid.delete_user("user", "xxx-yyy-zzz")
+        assert res == mocked_response
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_delete_user_failure(self, requests_mock: Mocker, rapid: Rapid):
+        mocked_response = {"data": "dummy"}
+        requests_mock.delete(
+            f"{RAPID_URL}/user", json=mocked_response, status_code=400
+        )
+        with pytest.raises(SubjectNotFoundException):
+            rapid.delete_user("user", "xxx-yyy-zzz")
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_list_subjects(self, requests_mock: Mocker, rapid: Rapid):
+        expected = {"response": "dummy"}
+        requests_mock.get(f"{RAPID_URL}/subjects", json=expected)
+
+        res = rapid.list_subjects()
+        assert res == expected
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_list_layers(self, requests_mock: Mocker, rapid: Rapid):
+        expected = {"response": "dummy"}
+        requests_mock.get(f"{RAPID_URL}/layers", json=expected)
+
+        res = rapid.list_layers()
+        assert res == expected
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_list_protected_domains(self, requests_mock: Mocker, rapid: Rapid):
+        expected = {"response": "dummy"}
+        requests_mock.get(f"{RAPID_URL}/protected_domains", json=expected)
+
+        res = rapid.list_protected_domains()
+        assert res == expected
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_delete_dataset_success(self, requests_mock: Mocker, rapid: Rapid):
+        layer = "raw"
+        domain = "test_domain"
+        dataset = "test_dataset"
+        mocked_response = {'details': '{dataset} has been deleted.'}
+        requests_mock.delete(
+            f"{RAPID_URL}/datasets/{layer}/{domain}/{dataset}",
+              json=mocked_response, 
+              status_code=202
+        )
+        res = rapid.delete_dataset(layer, domain, dataset)
+        assert res == mocked_response
+
+    @pytest.mark.usefixtures("requests_mock", "rapid")
+    def test_delete_dataset_failure(self, requests_mock: Mocker, rapid: Rapid):
```

Comment:
Will need to test incorrect permissions for deleting dataset if user doesn't have write permissions.

Reply:
Test added.